### PR TITLE
Port testing to Python 3

### PR DESCRIPTION
**Pull request by chadwhitacre**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3621

Fixes: ticket:3621

### CODE REVIEW

**Comment by itamarst** [comment]
Is this still WIP?

**Comment by chadwhitacre** [comment]
Nope. Needed a CI cleanse. Should be good now, will merge next time I'm here unless someone objects.

**Review by itamarst — CHANGES_REQUESTED** [review]
1. You need to add this to `_python3.py`.
2. You haven't ported `allmydata.test.test_testing`.

**Comment by chadwhitacre** [comment]
Thanks @itamarst, should be in a better spot now.

File: src/allmydata/util/_python3.py [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "allmydata.storage.shares",
     "allmydata.test.no_network",
     "allmydata.test.mutable.util",
+    "allmydata.testing",
```
Comment (itamarst) [inline src/allmydata/util/_python3.py:90]: This should probably be `allmydata.testing.web`.

**Review by itamarst — APPROVED** [review]

**Review by exarkun — APPROVED** [review]
Apart from itamar's comment on the new entry in the ported-modules list (which I agree with), this looks good to me.  Please address that and then merge.

**Pull request merged**